### ISSUE TITLE
CustomImportControl bad document examples that could lead to false positive on 'special group'

### ISSUE THREAD

**Issue by Vladlis** [checkstyle/checkstyle#3590]
```
vlad@vlad-W65-67SB ~/Documents $ java -jar checkstyle-7.3-all.jar -c my_check.xml Input.java
Starting audit...
[WARN] /home/vlad/Documents/Input.java:4: Import statement for 'com.github.checkstyle.publishers.SourceforgeRssPublisher' is in the wrong order. Should be in the 'SPECIAL_IMPORTS' group, expecting not assigned imports on this line. [CustomImportOrder]
Audit done.
vlad@vlad-W65-67SB ~/Documents $ cat Input.java 
package com.puppycrawl.tools.checkstyle.checks.imports;

import com.github.checkstyle.publishers.MailingListPublisher;
import com.github.checkstyle.publishers.SourceforgeRssPublisher;
import com.github.checkstyle.publishers.TwitterPublisher;
import com.github.checkstyle.publishers.XdocPublisher;

public class Input { }

vlad@vlad-W65-67SB ~/Documents $ cat my_check.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>

    <module name="TreeWalker">
         <module name="CustomImportOrder">
             <property name="customImportOrderRules" value="STATIC###STANDARD_JAVA_PACKAGE###SPECIAL_IMPORTS"/>
             <property name="specialImportsRegExp" value="org"/>
         </module>
    </module>
</module>
```
Expected: no violations.

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39597513-customimportcontrol-false-positive-on-special-group?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github).

**Comment by MEZk** [comment]
@Vladlis @romani 

It is not a false positive since @Vladlis specified regexp for specialImportsRegExp which matches the substring in SourceforgeRssPublisher. The correct regexp should be ^org.*$

**Comment by romani** [comment]
Yes, lets add it to xdoc for good example of regexp that is ready to use and not errorprone

**Comment by romani** [comment]
fix is merged
